includes/ziti/model_map.h

```diff
@@ -80,6 +78,35 @@ for (var(e,line) = model_map_iterator(map), k = model_map_it_key(var(e,line)), v
 
 #define MODEL_MAP_FOR(it, m) for(model_map_iter it = model_map_iterator(&(m)); (it) != NULL; (it) = model_map_it_next(it))
 
+typedef struct model_list_s {
+    struct model_list_impl_s *impl;
+} model_list;
+
+typedef void *model_list_iter;
+
+ZITI_FUNC size_t model_list_size(const model_list *l);
+
+ZITI_FUNC void model_list_append(model_list *l, void *el);
+
+ZITI_FUNC void *model_list_pop(model_list *l);
+
+ZITI_FUNC void *model_list_head(const model_list *l);
+
+ZITI_FUNC model_list_iter model_list_iterator(model_list *l);
+
+ZITI_FUNC model_list_iter model_list_it_next(model_list_iter it);
+
+ZITI_FUNC model_list_iter model_list_it_remove(model_list_iter it);
+
+ZITI_FUNC void *model_list_it_element(model_list_iter it);
+
+#define MODEL_LIST_FOR(it, m) for(model_list_iter it = model_list_iterator(&(m)); (it) != NULL; (it) = model_list_it_next(it))
+
+#define MODEL_LIST_FOREACH(el, list) for(model_list_iter it = model_list_iterator(&(list)); \
+(it) != NULL && (((el) = model_list_it_element(it)), true);                                 \
+(it) = model_list_it_next(it))
+
```

Comment:
is there no 'free' for the whole model_list? would i be expected to iterate and remove them all on my own? something else?

Reply:
added `model_list_clear()`